exchanges/okx/ws_requests.go

```diff
@@ -0,0 +1,344 @@
+		}
+		if args[x].ClientOrderID == "" && args[x].OrderID == "" {
+			return nil, order.ErrOrderIDNotSet
+		}
+		if args[x].NewQuantity <= 0 && args[x].NewPrice <= 0 {
+			return nil, errInvalidNewSizeOrPriceInformation
+		}
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*OrderData
+	return resp, ok.SendAuthenticatedWebsocketRequest(ctx, amendMultipleOrdersEPL, id, "batch-amend-orders", args, &resp)
+}
+
+// WSMassCancelOrders cancels all MMP pending orders of an instrument family. Only applicable to Option in Portfolio Margin mode, and MMP privilege is required.
+func (ok *Okx) WSMassCancelOrders(ctx context.Context, args []CancelMassReqParam) error {
+	if len(args) == 0 {
+		return fmt.Errorf("%T: %w", args, order.ErrSubmissionIsNil)
+	}
+
+	for x := range args {
+		if args[x].InstrumentType == "" {
+			return fmt.Errorf("%w, instrument type can not be empty", errInvalidInstrumentType)
+		}
+		if args[x].InstrumentFamily == "" {
+			return errInstrumentFamilyRequired
+		}
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resps []*struct {
+		Result bool `json:"result"`
+	}
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, amendOrderEPL, id, "mass-cancel", args, &resps); err != nil {
+		return err
+	}
+
+	resp, err := singleItem(resps)
+	if err != nil {
+		return err
+	}
+
+	if !resp.Result {
+		return errMassCancelFailed
+	}
+
+	return nil
+}
+
+// WSPlaceSpreadOrder submits a spread order
+func (ok *Okx) WSPlaceSpreadOrder(ctx context.Context, arg *SpreadOrderParam) (*SpreadOrderResponse, error) {
+	if err := arg.Validate(); err != nil {
+		return nil, err
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*SpreadOrderResponse
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, placeSpreadOrderEPL, id, "sprd-order", []SpreadOrderParam{*arg}, &resp); err != nil {
+		return nil, err
+	}
+
+	return singleItem(resp)
+}
+
+// WSAmendSpreadOrder amends a spread order
+func (ok *Okx) WSAmendSpreadOrder(ctx context.Context, arg *AmendSpreadOrderParam) (*SpreadOrderResponse, error) {
+	if arg == nil {
+		return nil, fmt.Errorf("%T: %w", arg, common.ErrNilPointer)
+	}
+	if arg.OrderID == "" && arg.ClientOrderID == "" {
+		return nil, order.ErrOrderIDNotSet
+	}
+	if arg.NewPrice == 0 && arg.NewSize == 0 {
+		return nil, errSizeOrPriceIsRequired
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*SpreadOrderResponse
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, amendSpreadOrderEPL, id, "sprd-amend-order", []AmendSpreadOrderParam{*arg}, &resp); err != nil {
+		return nil, err
+	}
+
+	return singleItem(resp)
+}
+
+// WsCancelSpreadOrder cancels an incomplete spread order through the websocket connection.
+func (ok *Okx) WsCancelSpreadOrder(ctx context.Context, orderID, clientOrderID string) (*SpreadOrderResponse, error) {
```

Comment:
```suggestion
func (ok *Okx) WSCancelSpreadOrder(ctx context.Context, orderID, clientOrderID string) (*SpreadOrderResponse, error) {
```
To be consistent with our naming scheme